src/csharp/Extensions/Microsoft.Spark.Extensions.DotNet.Interactive/PackageResolver.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.DotNet.Interactive;
+using Microsoft.DotNet.Interactive.Utility;
+
+namespace Microsoft.Spark.Extensions.DotNet.Interactive
+{
+    internal class PackageResolver : IPackageResolver
+    {
+        private readonly HashSet<string> _filesCopied = new HashSet<string>();
+
+        /// <summary>
+        /// Return the delta of the list of packages that have been introduced
+        /// since the last call.
+        /// </summary>
+        /// <returns>The delta of the list of packages.</returns>
+        public IEnumerable<ResolvedNuGetPackage> GetPackagesToCopy()
+        {
+            PackageRestoreContext restoreContext =
+                ((ISupportNuget)KernelInvocationContext.Current.HandlingKernel)
```

Comment:
should `ISupportNuget` be passed into the constructor of this class? In that case, is this class also testable now (not sure how hard it is to mock `ResolvedPackageReference`)? And I think this function can move to PackageHelper? (I don't get the abstraction layers of `PackageResolver` vs `PackageHelper`)

Reply:
Originally moved out logic from `PackageHelper` into `PackageResolver` to do unit tests.  Couldn't mock `ISupportNuget`, because `PackageRestoreContext.ResolvedPackageReferences` isn't a virtual property.

Moved logic back into `PackageHelper` and instead wrapping the `ISupportNuget` ... `PackageRestoreContext.ResolvedPackageReferences` call.